internal/twirptest/service.twirp.go

```diff
@@ -565,33 +540,68 @@ func twirpErrorFromIntermediary(status int, msg string, bodyOrLocation string) t
 	}
 	return twerr
 }
+
 func isHTTPRedirect(status int) bool {
 	return status >= 300 && status <= 399
 }
 
-// wrappedError implements the github.com/pkg/errors.Causer interface, allowing errors to be
-// examined for their root cause.
+// wrapInternal wraps an error with a prefix as an Internal error.
+// The original error cause is accessible by github.com/pkg/errors.Cause.
+func wrapInternal(err error, prefix string) twirp.Error {
+	return twirp.InternalErrorWith(&wrappedError{prefix: prefix, cause: err})
+}
+
 type wrappedError struct {
-	msg   string
-	cause error
+	prefix string
+	cause  error
+}
+
+func (e *wrappedError) Cause() error  { return e.cause }
+func (e *wrappedError) Error() string { return e.prefix + ": " + e.cause.Error() }
+
+// handlePanics makes sure that panics during rpc methods respond with Twirp Interna
+// error responses (status 500) and trigger hooks with the panic wrapped as an error.
+// The panic is not intercepted and should be handled in middleware to avoid empty responses.
```

Comment:
I'd phrase this differently. The panic is definitely intercepted, we just re-raise it.

Empty responses are kind of a separate issue that we should be able to deal with by flushing correctly.

Reply:
Yes. I'll phrase it differently to be more clear. I found the empty response issue while working on this, it was not as clear to me before.

Reply:
For reference, the empty responses issue is tracked here https://github.com/twitchtv/twirp/issues/163